control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -412,45 +414,71 @@ impl ResourcePublishing for OperationGuardArc<VolumeSpec> {
             target_cfg.config().clone(),
         ));
         let spec_clone = self.start_update(registry, &state, operation).await?;
-        let result = specs.nexus(volume_target.nexus()).await;
-        let mut older_nexus = self
-            .validate_update_step(registry, result, &spec_clone)
-            .await?;
 
-        let older_nexus_id = older_nexus.uuid().clone();
-        // shutdown the older nexus before newer nexus creation.
-        let result = older_nexus
-            .shutdown(registry, &ShutdownNexus::new(older_nexus_id))
-            .await;
+        let result =
+            get_healthy_volume_replicas(&spec_clone, target_cfg.target().node(), registry).await;
         self.validate_update_step(registry, result, &spec_clone)
             .await?;
 
-        // Create a Nexus on the requested or auto-selected node
-        let result = specs
-            .volume_create_nexus(registry, &target_cfg, &spec_clone)
-            .await;
-        let (mut nexus, nexus_state) = self
+        let result = specs.nexus(volume_target.nexus()).await;
+        let mut older_nexus = self
             .validate_update_step(registry, result, &spec_clone)
             .await?;
 
-        // Share the Nexus
-        let result = match nexus
-            .share(
-                registry,
-                &ShareNexus::from((&nexus_state, None, request.share)),
-            )
+        let mut move_nexus = true;
+
+        if let Ok(nexus) = registry
+            .get_node_nexus(&older_nexus.as_ref().node, &older_nexus.as_ref().uuid)
             .await
         {
-            Ok(_) => Ok(()),
-            Err(error) => {
-                // Since we failed to share, we'll revert back to the previous state.
-                // If we fail to do this inline, the reconcilers will pick up the slack.
-                nexus
-                    .destroy(registry, &DestroyNexus::from(nexus_state).with_disown_all())
-                    .await
-                    .ok();
-                Err(error)
+            if nexus.status == NexusStatus::Online || nexus.status == NexusStatus::Degraded {
+                tracing::debug!(
```

Comment:
Can it be info? I think this should be logged by default.

Reply:
I think should be info too.
Also, I think we need to check that the nexus spec is not set to shutdown?